client/src/components/CollegeShortlist.js

```diff
@@ -0,0 +1,110 @@
+import React, { Component } from 'react';
+import { Menu, Label, Card, Button, Icon } from 'semantic-ui-react';
+import CollegeShortlistModal from './CollegeShortlistModal';
+
+const STUDENT = "STUDENT"
+
+export default class CollegeShortlist extends Component {
+    constructor(props) {
+        super(props)
+        this.state = {
+            activeItem: "shortlist",
+            collegeModalOpen: false,
+            collegeShortlist: [],
+        }
+        this.openCollegeModal = this.openCollegeModal.bind(this);
+        this.closeCollegeModal = this.closeCollegeModal.bind(this);
+        this.addToShortlist = this.addToShortlist.bind(this);
+    }
+
+    openCollegeModal() {
+        this.setState({
+            collegeModalOpen: true
+        })
+    }
+    closeCollegeModal() {
+        this.setState({
+            collegeModalOpen: false
+        }, () => {
+            this.props.refreshProfile(STUDENT, this.props.details._id)
+        })
+    }
+    addToShortlist(newCollege) {        
+        let currentCollegeShortlist = this.props.details.collegeShortlist
+        currentCollegeShortlist.push(newCollege) 
+        this.setState({
+            shortlist: currentCollegeShortlist
+        })
+        // let currentCollegeShortlist = this.state.collegeShortlist
+        this.state.collegeShortlist.push(newCollege);
+    }
+
+    render() {
+        const details = this.props.details;
+        const isViewOnly = this.props.isViewOnly;
+        // if (details.collegeShortlist.size > 0) {
+            
+        //     this.collegeShortlist = details.collegeShortlist.map(e => e.name).join(', ');
+        // }
+        return (
+            <div>
+            
+            <Menu secondary stackable>
+                <Menu.Item
+                    id='shortlist'
+                    name='College Shortlist'
+                    active={this.state.activeItem === 'shortlist'}
+                    onClick={this.handleMenuClick}
+                >
+                    College Shortlist
+                    {   (this.state.shortlist !== []) && 
+                         <Label color='teal'>{details.collegeShortlist.length}</Label>
+                        //  <Label color='teal'>{this.state.collegeShortlist.length}</Label>
+                    }
+                </Menu.Item>
+                
+            </Menu>
+            {this.state.activeItem === 'shortlist' &&
+                <div style={{paddingLeft: 13, paddingRight: 13}}>
+                    <Card fluid>
+                    <Card.Content>
+                    <Card.Header>{details.name || 'Unavailable'}</Card.Header>
+                    {/* <Card.Description>Colleges: {this.state.collegeShortlist[0] || 'Unavailable'} */}
+                    <Card.Description>Colleges: {details.collegeShortlist.map(e => 
```

Comment:
This is probably the reason your state related version didn't work - You're constructing the shortlist based on the props only, when you are generally making changes to the state to try and trigger updates

Reply:
It might be worth only working with state for this Component
You might need to add a component will / did mount to set the initial state, but it will make other changes easier and will be simpler to reflect, because if a state is referenced in render method and it changes, the component will re-render automatically